File: memory-common/src/main/java/com/memory/annotation/MemberId.java
package com.memory.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation to inject the memberId extracted from the JWT token.
 * This annotation can be used on method parameters to receive the memberId.
 */
@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface MemberId {
}

File: memory-api/src/main/kotlin/com/memory/annotation/MemberId.kt
package com.memory.annotation

/**
 * Annotation to inject the memberId extracted from the JWT token.
 * This annotation can be used on method parameters to receive the memberId.
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class MemberId 